Tools/wasm/emscripten/__main__.py

```diff
@@ -215,12 +215,29 @@ def configure_emscripten_python(context, working_dir):
     exec_script = working_dir / "python.sh"
     exec_script.write_text(
         dedent(
-            f"""\
+            """\
             #!/bin/sh
 
+            # Macs come with a defective fork of coreutils so feature detect and
+            # work around it.
+            if which grealpath > /dev/null; then
+                # It has brew installed gnu core utils, use that
+                REALPATH="grealpath -s"
+            elif which readlink > /dev/null && realpath --version | grep GNU > /dev/null; then
+                # realpath points to GNU realpath so use it.
+                REALPATH="realpath -s"
+            else
+                # Shim for macs without GNU coreutils
+                abs_path () {
+                    echo "$(cd $(dirname "$1");pwd)/$(basename "$2")"
```

Comment:
Should this be:
```suggestion
                    echo "$(cd $(dirname "$1");pwd)/$(basename "$1")"
```
?

However, even then, if I try to run a `hello.py` (that contains `print("hello, world")`) that is in the root directory of the project, I get:
```
(venv) rkm@eunectes wasm32-emscripten % ./python.sh hello.py
realpath: illegal option -- -
usage: realpath [-q] [path ...]
/Users/rkm/projects/python/cpython/cross-build/wasm32-emscripten/python.sh: can't open file '//hello.py': [Errno 44] No such file or directory
```
Am I misunderstanding the intended usage?

Reply:
Yeah... This is currently the expected output because of other bugs ("limitations") in `node_entry.mjs`. I wasn't planning to fix them in this PR because it's not relevant to the description.

Basically though the problems are:
1. Working directory not mounted into emscripten file system so it can't see `some_file.py`
2. Working directory not set correctly in emscripten so even if it were mounted it would look in the wrong place
3. Environment variables not propagated so in particular `HOME` is weird

Before then it won't work all that well like the python cli but with these changes it works pretty okay. I'm still also waiting on https://github.com/emscripten-core/emscripten/pull/22925 to get merged so that named pipes won't cause chaos.

Maybe I should make a separate PR and move the bsd realpath fixes and these three fixes there.

Reply:
https://github.com/python/cpython/pull/127506